src/sage/combinat/affine_permutation.py

```diff
@@ -77,13 +76,13 @@ def __init__(self, parent, lst, check=True):
         """
         if check:
             lst = [ZZ(val) for val in lst]
-        self.k = parent.k
+        self.k = ZZ(parent.k)
```

Comment:
Is this `ZZ` call necessary? In the `__init__` for `AffinePermutationGroupGeneric`, you have `self.k = ZZ(ct.n)` so I think `parent.k` will always be a Sage integer unless there's some other code path that bypasses the existing cast.

Even in Sage 10.5 I don't see how to construct this object in a way that you don't get a Sage integer here.